.github/workflows/image-multiarch.yaml

```diff
@@ -0,0 +1,67 @@
+name: Build & push multi-arch image
+
+on:
+  workflow_call:
+    secrets:
+      GH_TOKEN:
+        description: "GitHub token"
+        required: true
+      ECR_REGISTRY:
+        description: "Repository to push the image"
+        required: true
+    inputs:
+      service:
+        description: "Name to tag the image"
+        type: string
+        required: true
+      build-args:
+        description: "Build args"
+        type: "string"
+        required: false
+      context:
+        description: "Build context"
+        type: string
+        required: false
+        default: "."
+      labels:
+        description: "Image labels"
+        type: string
+        required: false
+jobs:
+  build:
+    name: Build and push image
+    runs-on: [ self-hosted, bear-ephemeral ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker context for Buildx
+        run: docker context create builders
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          endpoint: builders
+      - name: Build and push images
+        uses: docker/build-push-action@v5
+        with:
+          build-args: ${{ inputs.build-args }}
+          context: ${{ inputs.context }}
+          labels: ${{ inputs.labels }}
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.ECR_REGISTRY }}/${{ inputs.service }}:${{ github.run_id }}
+          secrets: |
+            "github_token=${{ secrets.GH_TOKEN }}"
+      - name: Push latest image
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        uses: docker/build-push-action@v5
+        with:
+          build-args: ${{ inputs.build-args }}
+          context: ${{ inputs.context }}
+          labels: ${{ inputs.labels }}
+          platforms: linux/amd64,linux/arm64
+          push: true
+          secrets: |
+            "github_token=${{ secrets.GH_TOKEN }}"
```

Comment:
My only concern here would be in the case some image has additional secrets, we probably would need to provide a way to provide "arbitrary" secrets and not just this hardcoded one.
I'm also not sure if ALL images need a github token, and this could be a small security problem (only expose the token for images that actually need it).

Reply:
Absolutely agree, fixed !